apps/react/src/features/anime/components/AnimeImage/AnimeImage.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC, memo, useState } from 'react';
+import { Box, Button, Dialog, DialogContent } from '@mui/material';
+
+import styles from './AnimeImage.module.css';
+
+/** Anime image props. */
+type Props = Readonly<{
+
+	/** Image source. */
+	imageSrc: string;
+
+	/** Image alternative text. */
+	imageAlt: string;
+}>;
+
+/** Anime image component. */
+const AnimeImageComponent: FC<Props> = ({ imageSrc, imageAlt }: Props) => {
+	const [openImagePopup, setOpenImagePopup] = useState(false);
+
+	return <>
+		<Box component='img' className={styles['anime-image']} src={imageSrc} alt={imageAlt} />
+
+		<Button className={styles['anime-image-button']} tabIndex={1} onClick={() => setOpenImagePopup(true)}>
+			View full image
+		</Button>
+
+		<Dialog open={openImagePopup} onClose={() => setOpenImagePopup(false)}>
```

Comment:
`openImagePopup` sounds like a function name, let's improve it

```suggestion
		<Dialog open={isPopupOpened} onClose={() => setOpenImagePopup(false)}>
```